app/renderer/ui/containers/WalletForm.tsx

```diff
@@ -0,0 +1,390 @@
+  WalletSeedBackup,
+  WalletSeedConfirmation,
+  WalletSeedTypeSelection,
+  Welcome
+} from "app/renderer/ui/components/walletForm/steps"
+
+/**
+ * Props that match redux store state
+ *
+ * @export
+ * @interface StateProps
+ */
+export interface StateProps {
+  wallets: Wallets,
+}
+
+/**
+ * Props that match redux actions
+ *
+ * @export
+ * @interface DispatchProps
+ */
+export interface DispatchProps {
+  actions: any
+}
+
+/**
+ * Props that contain configuration
+ *
+ * @interface Props
+ */
+interface Props {
+  services: any
+}
+
+/**
+ * Function to map the dispatch of actions to props
+ * @param dispatch
+ */
+const mapDispatchToProps = (dispatch: Dispatch<IAction>): DispatchProps => {
+  return {
+    actions: bindActionCreators({ saveWallet }, dispatch)
+  }
+}
+
+/**
+ * Function to map store state to props
+ * @param state
+ */
+const mapStateToProps = (state: StoreState): StateProps => {
+  return ({
+    wallets: state.wallets,
+  })
+}
+
+export const PATHS = {
+  WELCOME: "/signup/welcome",
+  WALLET_SEED_TYPE_SELECTION: "/signup/wallet-seed-type-selection",
+  WALLET_SEED_BACKUP: "/signup/wallet-seed-backup",
+  WALLET_SEED_CONFIRMATION: "/signup/wallet-seed-confirmation",
+  WALLET_ENCRYPTION_PASSWORD: "/signup/wallet-encryption-password",
+  ADDRESS_GENERATION: "/signup/address-generation",
+  MAIN: "/main"
+}
+
+/**
+ * Wallet form container UI component
+ *
+ * @class WalletFormContainer
+ * @extends {(React.Component<RouteComponentProps<any> & Props & StateProps & DispatchProps>)}
+ */
+class WalletFormContainer extends
+  React.Component<RouteComponentProps<any> & Props & StateProps & DispatchProps> {
+
+  public state = {
+    confirmMnemonics: "",
+    mnemonics: "",
+    mnemonicsErrorMessage: "",
+    password: "",
+    passwordErrorMessage: "",
+    repeatPassword: "",
+    spinner: false,
+  }
+
+  /**
+   * method to handle react router navigation
+   *
+   * @private
+   * @memberof WalletFormContainer
+   */
+  private to = (string: string) => {
+    return () => this.props.history.replace(string)
+  }
+
+  /**
+   * method to render welcome step
+   *
+   * @private
+   * @memberof WalletFormContainer
+   */
+  private renderWelcome = () => <Welcome nextStep={this.to(PATHS.WALLET_SEED_TYPE_SELECTION)} />
+
+  /**
+   * method to render seed type selection step
+   *
+   * @private
+   * @memberof WalletFormContainer
+   */
+  private renderSeedTypeSelection = () => {
+    const steps = {
+      // TODO: add seed type options
+      newSeed: async () => {
+        if (!this.state.mnemonics) {
+          await this.setState({ spinner: true })
+          const mnemonicsResponse = await newMnemonics(this.props.services.apiClient)
+          await this.setState({ spinner: false })
+          if (mnemonicsResponse.kind === "SUCCESS") {
+            const newState = { mnemonics: mnemonicsResponse.mnemonics }
+            await this.setState(newState)
+          } else {
+            // TODO: improve error handler
```

Comment:
I wouldn't left this as a TODO, or at least we should open an issue and link this comment to that issue